ir/visitor.h

```diff
@@ -57,7 +57,11 @@ class Visitor {
     virtual ~Visitor() = default;
 
     mutable cstring internalName;
+    // Some visitors are created and applied by other visitors.
+    // This field keeps track of the caller.
+    const Visitor* called_by = nullptr;
 
+    const Visitor& setCalledBy(const Visitor* visitor) { called_by = visitor; return *this; }
```

Comment:
This addition seems completely unnecessary -- you could just call `apply` with the context as a second argument .  You then just need a simple search up the context.

Reply:
Didn't realize you can thread context across visitors. You can file an issue to make this change if you think it's worth the effort